JDBC/DatabasesPractice/src/main/java/com/laba/solvd/database/DAO/StudentDAO.java

```diff
@@ -0,0 +1,188 @@
+package com.laba.solvd.database.DAO;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+
+import com.laba.solvd.database.Model.Gym;
+import com.laba.solvd.database.Model.Professor;
+import com.laba.solvd.database.Model.ResearchLab;
+import com.laba.solvd.database.Model.SportsTeam;
+import com.laba.solvd.database.Model.Student;
+import com.laba.solvd.database.Model.UniClass;
+
+public class StudentDAO implements IDAO<Student>, ICreatableWithRelationship<Student, Gym>{
+
+	@Override
+	public void create(Student student, Gym gym) {
+		try (Connection connection = (Connection) ConnectionManager.getSQLConnection().get();) {
+			PreparedStatement ps = connection.prepareStatement("INSERT INTO Student(name, yearJoined, major, Gym_id) VALUES(?, ?, ?, ?)");
+			ps.setString(1, student.getName());
+			ps.setInt(2, student.getYearJoined());
+			ps.setString(3, student.getMajor());
+			ps.setInt(4, gym.getId());
+			ps.executeUpdate();
+		} catch (SQLException | InterruptedException | ExecutionException e) {
+			e.printStackTrace();
+		}
+	}
+	
+	@Override
+	public List<Student> selectAll() {
+		List<Student> students = new ArrayList<>();	
+		try (Connection connection = (Connection) ConnectionManager.getSQLConnection().get();) {
+			PreparedStatement ps = connection.prepareStatement(String.format(GET_ALL, "student"));
+			ResultSet resultSet = ps.executeQuery();
+			while (resultSet.next()) {
+				Student student = new Student();
+				student.setId(resultSet.getInt("id"));
+				student.setMajor(resultSet.getString("major"));
+				student.setName(resultSet.getString("name"));
+				student.setYearJoined(resultSet.getInt("yearJoined")); 
+				student.setGym(new GymDAO().selectById(resultSet.getInt("Gym_id")));
+				students.add(student);
+			}
+		} catch (SQLException | InterruptedException | ExecutionException e) {
+			e.printStackTrace();
+		}
+		return students;
+	}
+	
+	@Override
+	public Student selectById(int Id) {
+		Student student = new Student();
+		try (Connection connection = (Connection) ConnectionManager.getSQLConnection().get();) {
+			PreparedStatement psStudentFields = connection.prepareStatement(String.format(GET_BY_ID, "student"));
+			psStudentFields.setInt(1, Id);
+			ResultSet StudentFields = psStudentFields.executeQuery();

```

Comment:
Just like Adela mentioned earlier, you need to include StudentFields.next() to make the cursor point to the first row. I was getting an error without adding that. I am not sure how its working for you.